sdk/search/Azure.Search/src/SearchClientOptions.cs

```diff
@@ -92,10 +94,31 @@ internal HttpPipeline Build(SearchApiKeyCredential credential)
         /// </summary>
         private void AddLoggingHeaders()
         {
-            Diagnostics.LoggedHeaderNames.Add("x-ms-version");
-            Diagnostics.LoggedHeaderNames.Add("OData-version");
+            Diagnostics.LoggedHeaderNames.Add("Accept");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Allow-Credentials");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Allow-Headers");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Allow-Methods");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Allow-Origin");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Expose-Headers");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Max-Age");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Request-Headers");
+            Diagnostics.LoggedHeaderNames.Add("Access-Control-Request-Method");
+            Diagnostics.LoggedHeaderNames.Add("client-request-id");
+            Diagnostics.LoggedHeaderNames.Add("Content-Type");
             Diagnostics.LoggedHeaderNames.Add("elapsed-time");
-            // TODO: Add the rest...
+            Diagnostics.LoggedHeaderNames.Add("If-Match");
+            Diagnostics.LoggedHeaderNames.Add("If-None-Match");
+            Diagnostics.LoggedHeaderNames.Add("Location");
+            Diagnostics.LoggedHeaderNames.Add("OData-MaxVersion");
+            Diagnostics.LoggedHeaderNames.Add("OData-Version");
+            Diagnostics.LoggedHeaderNames.Add("Origin");
+            Diagnostics.LoggedHeaderNames.Add("Prefer");
+            Diagnostics.LoggedHeaderNames.Add("request-id");
+            Diagnostics.LoggedHeaderNames.Add("return-client-request-id");
+            Diagnostics.LoggedHeaderNames.Add("throttle-reason");
+            Diagnostics.LoggedHeaderNames.Add("User-Agent");
+            Diagnostics.LoggedHeaderNames.Add("x-ms-client-request-id");
```

Comment:
Aren't some of these in default options?

Reply:
Probably - I'll go double check.  Might be worth adding a unit test that these are always unique too.

Reply:
Fixed and added a unit test